fastify4ts/package.json

```diff
@@ -0,0 +1,50 @@
+{
+  "name": "@contrast/fastify-ts-test-bench",
+  "private": true,
+  "version": "1.0.0",
+  "description": "Intentionally vulnerable Fastify application written in Typescript",
+  "keywords": [],
+  "author": "Contrast Security <nodejs@contrastsecurity.com>",
+  "license": "UNLICENSED",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Contrast-Security-OSS/NodeTestBenches.git"
+  },
+  "engines": {
+    "node": ">=12.13.0"
+  },
+  "bugs": {
+    "url": "https://github.com/Contrast-Security-OSS/NodeTestBenches/issues"
+  },
+  "homepage": "https://github.com/Contrast-Security-OSS/NodeTestBenches/tree/main/fastify3#readme",
+  "main": "index.js",
+  "scripts": {
+    "contrast": "DEBUG=contrast:* node-contrast server.js",
+    "dev": "nodemon build/server.js",
+    "start": "node build/server.js",
+    "build": "rm -rf build/ && tsc -p tsconfig.json && cp -r public/ build/public/ && cp -r view/ build/view/",
```

Comment:
I have an idea for a slightly more consolidated folder structure:

rather than nest _everything_ in `build` folder, we can use a `src/` and adjacent `lib/` structure, similar to our mono-repo modules.

```suggestion
    "build": "tsc --build src/tsconfig.json",
```

move _server.ts_, _tsconfig.json_, and the _routes_/ and _utils_/ directories to _src_.

update _tsconfig.json_ with:
```json
{
  "compilerOptions": {
    "outDir": "../lib"
  }
}
```

and finally, a couple lines in need to change:
```ts
    // setup ejs renderer
    // god damn point-of-view doesnt support layouts like every other
    // view front-end in node.
    app.register(view, {
      engine: {
        ejs: require('ejs')
      },
      templates: path.join(__dirname, '..', 'view'),
      includeViewExtension: true // dont want to write .ejs every time
    });

    // ...

    // setup public assets
    app.register(require('@fastify/static'), {
      root: path.join(__dirname, '..', 'public'),
      prefix: '/assets/'
    });
```

Reply:
here's a patch file that accomplished the above:
```diff
diff --git c/fastify4ts/.gitignore i/fastify4ts/.gitignore
index 89d3f92..f36fa63 100644
--- c/fastify4ts/.gitignore
+++ i/fastify4ts/.gitignore
@@ -1,2 +1,3 @@
+lib/
 uploads/*
 !uploads/.gitkeep
diff --git c/fastify4ts/package.json i/fastify4ts/package.json
index 6b9b1cc..1445455 100644
--- c/fastify4ts/package.json
+++ i/fastify4ts/package.json
@@ -20,9 +20,9 @@
   "main": "index.js",
   "scripts": {
     "contrast": "DEBUG=contrast:* node-contrast server.js",
-    "dev": "nodemon build/server.js",
-    "start": "node build/server.js",
-    "build": "rm -rf build/ && tsc -p tsconfig.json && cp -r public/ build/public/ && cp -r view/ build/view/",
+    "dev": "nodemon lib/server.js",
+    "start": "node lib/server.js",
+    "build": "rm -rf build/ && tsc --build src/tsconfig.json",
     "lint": "eslint --fix ."
   },
   "dependencies": {
diff --git c/fastify4ts/routes/autocompleteMissing.ts i/fastify4ts/src/routes/autocompleteMissing.ts
similarity index 100%
rename from fastify4ts/routes/autocompleteMissing.ts
rename to fastify4ts/src/routes/autocompleteMissing.ts
diff --git c/fastify4ts/routes/cacheControlsMissing.ts i/fastify4ts/src/routes/cacheControlsMissing.ts
similarity index 100%
rename from fastify4ts/routes/cacheControlsMissing.ts
rename to fastify4ts/src/routes/cacheControlsMissing.ts
diff --git c/fastify4ts/routes/class-validator.ts i/fastify4ts/src/routes/class-validator.ts
similarity index 100%
rename from fastify4ts/routes/class-validator.ts
rename to fastify4ts/src/routes/class-validator.ts
diff --git c/fastify4ts/routes/clickjackingControlsMissing.ts i/fastify4ts/src/routes/clickjackingControlsMissing.ts
similarity index 100%
rename from fastify4ts/routes/clickjackingControlsMissing.ts
rename to fastify4ts/src/routes/clickjackingControlsMissing.ts
diff --git c/fastify4ts/routes/cmdInjection.ts i/fastify4ts/src/routes/cmdInjection.ts
similarity index 100%
rename from fastify4ts/routes/cmdInjection.ts
rename to fastify4ts/src/routes/cmdInjection.ts
diff --git c/fastify4ts/routes/cmdInjectionSemanticChainedCommands.ts i/fastify4ts/src/routes/cmdInjectionSemanticChainedCommands.ts
similarity index 100%
rename from fastify4ts/routes/cmdInjectionSemanticChainedCommands.ts
rename to fastify4ts/src/routes/cmdInjectionSemanticChainedCommands.ts
diff --git c/fastify4ts/routes/cmdInjectionSemanticDangerousPaths.ts i/fastify4ts/src/routes/cmdInjectionSemanticDangerousPaths.ts
similarity index 100%
rename from fastify4ts/routes/cmdInjectionSemanticDangerousPaths.ts
rename to fastify4ts/src/routes/cmdInjectionSemanticDangerousPaths.ts
diff --git c/fastify4ts/routes/cookies.ts i/fastify4ts/src/routes/cookies.ts
similarity index 100%
rename from fastify4ts/routes/cookies.ts
rename to fastify4ts/src/routes/cookies.ts
diff --git c/fastify4ts/routes/crypto.ts i/fastify4ts/src/routes/crypto.ts
similarity index 100%
rename from fastify4ts/routes/crypto.ts
rename to fastify4ts/src/routes/crypto.ts
diff --git c/fastify4ts/routes/csp-header.ts i/fastify4ts/src/routes/csp-header.ts
similarity index 100%
rename from fastify4ts/routes/csp-header.ts
rename to fastify4ts/src/routes/csp-header.ts
diff --git c/fastify4ts/routes/cspHeaderInsecure.ts i/fastify4ts/src/routes/cspHeaderInsecure.ts
similarity index 100%
rename from fastify4ts/routes/cspHeaderInsecure.ts
rename to fastify4ts/src/routes/cspHeaderInsecure.ts
diff --git c/fastify4ts/routes/cspHeaderMissing.ts i/fastify4ts/src/routes/cspHeaderMissing.ts
similarity index 100%
rename from fastify4ts/routes/cspHeaderMissing.ts
rename to fastify4ts/src/routes/cspHeaderMissing.ts
diff --git c/fastify4ts/routes/header-injection.ts i/fastify4ts/src/routes/header-injection.ts
similarity index 100%
rename from fastify4ts/routes/header-injection.ts
rename to fastify4ts/src/routes/header-injection.ts
diff --git c/fastify4ts/routes/hstsHeaderMissing.ts i/fastify4ts/src/routes/hstsHeaderMissing.ts
similarity index 100%
rename from fastify4ts/routes/hstsHeaderMissing.ts
rename to fastify4ts/src/routes/hstsHeaderMissing.ts
diff --git c/fastify4ts/routes/index.ts i/fastify4ts/src/routes/index.ts
similarity index 100%
rename from fastify4ts/routes/index.ts
rename to fastify4ts/src/routes/index.ts
diff --git c/fastify4ts/routes/nosqlExpansion.ts i/fastify4ts/src/routes/nosqlExpansion.ts
similarity index 100%
rename from fastify4ts/routes/nosqlExpansion.ts
rename to fastify4ts/src/routes/nosqlExpansion.ts
diff --git c/fastify4ts/routes/nosqlInjection.ts i/fastify4ts/src/routes/nosqlInjection.ts
similarity index 100%
rename from fastify4ts/routes/nosqlInjection.ts
rename to fastify4ts/src/routes/nosqlInjection.ts
diff --git c/fastify4ts/routes/parampollution.ts i/fastify4ts/src/routes/parampollution.ts
similarity index 100%
rename from fastify4ts/routes/parampollution.ts
rename to fastify4ts/src/routes/parampollution.ts
diff --git c/fastify4ts/routes/pathTraversal.ts i/fastify4ts/src/routes/pathTraversal.ts
similarity index 100%
rename from fastify4ts/routes/pathTraversal.ts
rename to fastify4ts/src/routes/pathTraversal.ts
diff --git c/fastify4ts/routes/pathTraversalSemanticFileSecurityBypass.ts i/fastify4ts/src/routes/pathTraversalSemanticFileSecurityBypass.ts
similarity index 100%
rename from fastify4ts/routes/pathTraversalSemanticFileSecurityBypass.ts
rename to fastify4ts/src/routes/pathTraversalSemanticFileSecurityBypass.ts
diff --git c/fastify4ts/routes/sqlInjection.ts i/fastify4ts/src/routes/sqlInjection.ts
similarity index 100%
rename from fastify4ts/routes/sqlInjection.ts
rename to fastify4ts/src/routes/sqlInjection.ts
diff --git c/fastify4ts/routes/ssjs.ts i/fastify4ts/src/routes/ssjs.ts
similarity index 100%
rename from fastify4ts/routes/ssjs.ts
rename to fastify4ts/src/routes/ssjs.ts
diff --git c/fastify4ts/routes/ssrf.ts i/fastify4ts/src/routes/ssrf.ts
similarity index 100%
rename from fastify4ts/routes/ssrf.ts
rename to fastify4ts/src/routes/ssrf.ts
diff --git c/fastify4ts/routes/unsafeFileUpload.ts i/fastify4ts/src/routes/unsafeFileUpload.ts
similarity index 100%
rename from fastify4ts/routes/unsafeFileUpload.ts
rename to fastify4ts/src/routes/unsafeFileUpload.ts
diff --git c/fastify4ts/routes/untrustedDeserialization.ts i/fastify4ts/src/routes/untrustedDeserialization.ts
similarity index 100%
rename from fastify4ts/routes/untrustedDeserialization.ts
rename to fastify4ts/src/routes/untrustedDeserialization.ts
diff --git c/fastify4ts/routes/unvalidatedRedirect.ts i/fastify4ts/src/routes/unvalidatedRedirect.ts
similarity index 100%
rename from fastify4ts/routes/unvalidatedRedirect.ts
rename to fastify4ts/src/routes/unvalidatedRedirect.ts
diff --git c/fastify4ts/routes/xContentTypeHeaderMissing.ts i/fastify4ts/src/routes/xContentTypeHeaderMissing.ts
similarity index 100%
rename from fastify4ts/routes/xContentTypeHeaderMissing.ts
rename to fastify4ts/src/routes/xContentTypeHeaderMissing.ts
diff --git c/fastify4ts/routes/xpathInjection.ts i/fastify4ts/src/routes/xpathInjection.ts
similarity index 100%
rename from fastify4ts/routes/xpathInjection.ts
rename to fastify4ts/src/routes/xpathInjection.ts
diff --git c/fastify4ts/routes/xss.ts i/fastify4ts/src/routes/xss.ts
similarity index 100%
rename from fastify4ts/routes/xss.ts
rename to fastify4ts/src/routes/xss.ts
diff --git c/fastify4ts/routes/xssJSON.ts i/fastify4ts/src/routes/xssJSON.ts
similarity index 100%
rename from fastify4ts/routes/xssJSON.ts
rename to fastify4ts/src/routes/xssJSON.ts
diff --git c/fastify4ts/routes/xssProtectionHeaderDisabled.ts i/fastify4ts/src/routes/xssProtectionHeaderDisabled.ts
similarity index 100%
rename from fastify4ts/routes/xssProtectionHeaderDisabled.ts
rename to fastify4ts/src/routes/xssProtectionHeaderDisabled.ts
diff --git c/fastify4ts/routes/xssRenderedTemplates.ts i/fastify4ts/src/routes/xssRenderedTemplates.ts
similarity index 100%
rename from fastify4ts/routes/xssRenderedTemplates.ts
rename to fastify4ts/src/routes/xssRenderedTemplates.ts
diff --git c/fastify4ts/routes/xssStealthyRequire.ts i/fastify4ts/src/routes/xssStealthyRequire.ts
similarity index 100%
rename from fastify4ts/routes/xssStealthyRequire.ts
rename to fastify4ts/src/routes/xssStealthyRequire.ts
diff --git c/fastify4ts/routes/xxe.ts i/fastify4ts/src/routes/xxe.ts
similarity index 100%
rename from fastify4ts/routes/xxe.ts
rename to fastify4ts/src/routes/xxe.ts
diff --git c/fastify4ts/server.ts i/fastify4ts/src/server.ts
similarity index 96%
rename from fastify4ts/server.ts
rename to fastify4ts/src/server.ts
index 10aa782..fbe2a2d 100644
--- c/fastify4ts/server.ts
+++ i/fastify4ts/src/server.ts
@@ -47,7 +47,7 @@ const main = async () => {
       engine: {
         ejs: require('ejs')
       },
-      templates: `${__dirname}/view`,
+      templates: path.join(__dirname, '..', 'view'),
       includeViewExtension: true // dont want to write .ejs every time
     });
 
@@ -59,7 +59,7 @@ const main = async () => {
 
     // setup public assets
     app.register(require('@fastify/static'), {
-      root: path.join(__dirname, 'public'),
+      root: path.join(__dirname, '..', 'public'),
       prefix: '/assets/'
     });
 
diff --git c/fastify4ts/tsconfig.json i/fastify4ts/src/tsconfig.json
similarity index 99%
rename from fastify4ts/tsconfig.json
rename to fastify4ts/src/tsconfig.json
index ec68a5d..ac51444 100644
--- c/fastify4ts/tsconfig.json
+++ i/fastify4ts/src/tsconfig.json
@@ -51,7 +51,7 @@
     // "emitDeclarationOnly": true,                      /* Only output d.ts files and not JavaScript files. */
     // "sourceMap": true,                                /* Create source map files for emitted JavaScript files. */
     // "outFile": "./",                                  /* Specify a file that bundles all outputs into one JavaScript file. If 'declaration' is true, also designates a file that bundles all .d.ts output. */
-    "outDir": "./build",                                 /* Specify an output folder for all emitted files. */
+    "outDir": "../lib",                                  /* Specify an output folder for all emitted files. */
     // "removeComments": true,                           /* Disable emitting comments. */
     // "noEmit": true,                                   /* Disable emitting files from a compilation. */
     // "importHelpers": true,                            /* Allow importing helper functions from tslib once per project, instead of including them per-file. */
diff --git c/fastify4ts/utils/controllerFactory.ts i/fastify4ts/src/utils/controllerFactory.ts
similarity index 100%
rename from fastify4ts/utils/controllerFactory.ts
rename to fastify4ts/src/utils/controllerFactory.ts

```